02-peer-protocol.md

```diff
@@ -199,13 +199,23 @@ This message contains a transaction input.
     * [`u32`:`prevtx_vout`]
     * [`u32`:`sequence`]
 
+1. `tlv_stream`: `tx_add_input_tlvs`
```

Comment:
When setting this, we will need to set `prevtx_len` to `0` since it is part of the main fields of the message, which feels a bit hackish. An alternative design would be to remove `prevtx` from the main message fields and make it a TLV, which would make it more similar to `prevtxout`. I find this cleaner and more future-proof since other types of inputs won't include the whole `prevtx` (e.g. the previous funding output when using splicing), but it's more backwards-incompatible. But hey, it's now or never that we have the opportunity to make this cleaner, so it's worth studying that option!